examples/ensemble/stack_optimization.py

```diff
@@ -0,0 +1,101 @@
+"""
+=============================================
+Stack ensemble tuning on a regression problem
+=============================================
+
+This example shows the flexiblity provided when using `FeatureUnion`/`Pipeline`
+APIs together with the stack's meta estimator.
+
+It also shows the possibility of using stacked generalization with regression
+problems.
+
+"""
+import numpy as np
+
+from sklearn.ensemble import StackMetaEstimator
+from sklearn.pipeline import make_union, make_pipeline
+from sklearn.metrics import r2_score
+
+# base estimators
+from sklearn.svm import SVR
+from sklearn.linear_model import LinearRegression, Ridge, Lasso
+
+# dataset
+from sklearn.datasets import load_diabetes
+from sklearn.model_selection import train_test_split, ParameterGrid
+
+# parameters
+RANDOM_SEED = 17583
+
+X, y = load_diabetes(return_X_y=True)
+
+Xtrain, Xtest, ytrain, ytest = train_test_split(X, y, random_state=RANDOM_SEED)
+
+
+# Let's create some regressors using parameter grids
+configs = [{'constructor': SVR,
+            'parameter_grid': ParameterGrid({'C': [1, 1e3],
+                                             'kernel': ['rbf', 'linear']})},
+           {'constructor': Ridge,
+            'parameter_grid': ParameterGrid({'alpha': [1., .1, .01],
+                                             'normalize': [True, False],
+                                             'random_state': [RANDOM_SEED]})},
+           {'constructor': Lasso,
+            'parameter_grid': ParameterGrid({'alpha': [1., .1, .01],
+                                             'normalize': [True, False],
+                                             'random_state': [RANDOM_SEED]})}]
+
+base_regressors = []
+for config in configs:
+    base_regressors.extend([config['constructor'](**params)
+                            for params in list(config['parameter_grid'])])
+    print("Generated %d regressors of type %s"
+          % (len(config['parameter_grid']), config['constructor'].__name__))
+
+print("Total number of base regressors: %d" % len(base_regressors))
+
+# For each regressor, starting from a model made of all regressors, let's try
+# removing one by one and checking if the ensemble performs better. The models
+# will be combined with a LinearRegression. We'll fit them beforehand, so we
+# speed things up.
+
+combiner = LinearRegression()
+
+layer0_metas = [StackMetaEstimator(x) for x in base_regressors]
+initial_layer0 = make_union(*layer0_metas)
```

Comment:
this is an example specifically put together to show the *possible* flexibility of having `StackMetaEstimator` as a separate class. Saying it's complex because we have too many options just because we have a base class doesn't make sense to me. I can exclude the example or extend the docs even further (maybe split it into "basic" and "advanced" usages). It's specially simple if you see caioaao/scikit-learn#1.